src/main/java/org/sonarsource/sonarlint/ls/connected/ProjectBindingManager.java

```diff
@@ -389,7 +389,9 @@ private void stopUnusedEngines() {
   private void clearCachesAndStopEngine(String connectionId) {
     folderBindingCache.entrySet().removeIf(e -> e.getValue().isPresent() && e.getValue().get().getConnectionId().equals(connectionId));
     fileBindingCache.entrySet().removeIf(e -> e.getValue().isPresent() && e.getValue().get().getConnectionId().equals(connectionId));
-    tryStopServer(connectionId, connectedEngineCacheByConnectionId.remove(connectionId));
+    if (connectedEngineCacheByConnectionId.containsKey(connectionId)) {
```

Comment:
Already fixed in current `master` 😉

Reply:
Yes, I pulled this from `master`, not sure why it's in diff